local-volume/provisioner/pkg/common/common.go

```diff
@@ -130,20 +133,12 @@ func CreateLocalPVSpec(config *LocalPVConfig) *v1.PersistentVolume {
 // GetVolumeConfigFromMountedConfigMap gets volume configuration from the mounted configmap volume,
 func GetVolumeConfigFromMountedConfigMap() (map[string]MountConfig, error) {
 	storageClassMapping := make(map[string]string)
-	files, err := ioutil.ReadDir(ProvisionerConfigPath)
+	fileContents, err := ioutil.ReadFile(path.Join(ProvisionerConfigPath, ProvisonerStorageClassConfigFile))
```

Comment:
What if we keep an internal map of:
filename (key) -> value conversion function

Then this function can iterate over all the files, and call the appropriate conversion functions for each file, which will populate the internal config struct.

Reply:
I see, so the global map[string]func(), with values defined at the initialization?

Reply:
yup, it can just be statically defined as a const even.